### PR TITLE
DOC-1699/DOC-1702: Fix tables and formatting in Console metrics page & Remove "Database metrics not collected during resharding" article

### PR DESCRIPTION
**Pull request by cmilesb**
Staging link: https://docs.redis.com/staging/jira-doc-1699/rs/clusters/monitoring/console-metrics-definitions/

### CODE REVIEW

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 1]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
          /rs/clusters/monitoring/console-metrics-definitions.md
 
 ---
-The Redis Enterprise Software (RS) admin console shows metrics with information about the performance of the cluster, node, database, and shard.
-For Redis on Flash (ROF) databases, additional metrics are available.
+The Redis Enterprise Software admin console shows metrics with information about the performance of the cluster, node, database, and shard.
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:21]:
```suggestion
The Redis Enterprise Software admin console shows performance metrics for clusters, nodes, databases, and shards.
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 2]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
-| Expired objects/sec | Database, Shard | Number of expired objects per second (An expired object is an object with expired TTL that was deleted from the database) | This is more relevant for caching use cases.<br>Redis implements 2 expiry mechanisms:<br><br>1. Lazy expiry – on every access to an object Redis first checks whether the object has already been expired by looking at its TTL, and if ‘yes’ Redis deletes it<br><br>2. Active expiry – Redis constantly selects random keys and deletes them if it finds they were expired |
-| Fork CPU usage | Database, Shard | CPU usage of Redis child forks (for replication, BGSAVA...) |  |
+| Evicted objects/sec[^1] | Database, Shard | Number of objects evicted per second | Objects are evicted if:<br><ol><li>The database reaches its memory_limit</li><li>The [eviction policy]({{< relref "/rs/databases/memory-performance/eviction-policy" >}}) is not configured to `no-eviction`</li><li>The dataset keys are compliant with the selected eviction policy. For exmaple, with the `volatile-lru` eviction policy, Redis evicts expired keys.</li></ol> |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:29]:
```suggestion
| Evicted objects/sec[^1] | Database, Shard | Number of objects evicted per second | Objects are evicted if:<br><ol><li>The database reaches its memory_limit</li><li>The [eviction policy]({{< relref "/rs/databases/memory-performance/eviction-policy" >}}) is not configured to `no-eviction`</li><li>The dataset keys are compliant with the selected eviction policy. For example, with the `volatile-lru` eviction policy, Redis evicts expired keys.</li></ol> |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 3]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+| Fork CPU usage | Database, Shard | CPU usage of Redis child forks |  |
 | Free disk space | Cluster, Node | Remaining unused disk space |  |
 | Free ephemeral disk space | Cluster, Node | Remaining unused disk space on the ephemeral path |  |
 | Free RAM | Cluster, Node | RAM available for system use |  |
-| Hit ratio | Database, Shard | Calculated as number of ops on existing keys out of the total number of ops | `number_of_ops_on_exsiting_keys` / `total_ops` |
-| Incoming traffic | Cluster, Node, Database | Total incoming traffic (in bytes/sec) to the database |  |
-| Incoming traffic compressed | Active-Active | Total incoming compressed traffic (in bytes/sec) to the database |  |
-| Incoming traffic uncompressed | Active-Active | Total incoming uncompressed traffic (in bytes/sec) to the database |  |
-| Latency | Database | Latency per operation | The graph shows average, min, max and last values are also shown |
+| Hit ratio | Database, Shard | Calculated as number of operations on existing keys out of the total number of operations | `number_of_ops_on_exsiting_keys` / `total_ops` |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:35]:
```suggestion
| Hit ratio | Database, Shard | Calculated as number of operations on existing keys out of the total number of operations | `number_of_ops_on_existing_keys` / `total_ops` |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 4]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
-| Outgoing traffic | Cluster, Node, Database | Total outgoing traffic (in bytes/sec) from the database |  |
-| Pending writes max | Active-Active | Maximum number of writes queued |  |
-| Pending writes min | Active-Active | Minimum number of writes queued |  |
-| RAM fragmentation | Database, Shard | Ratio between the used (allocated) memory and the actual physical RAM that is used. |  |
-| Read misses/sec | Database, Shard | Number of read operations (per sec) on non-existing keys | This metric is more relevant for caching use cases. |
-| Reads latency | Database | Latency per read operation | The graph shows average, min, max and last values are also shown |
-| Reads/sec | Database | Number of total reads per second | For example: GET |
+| Other commands latency | Database | Latency per other command | The graph shows average, min, max and last latency values |
+| Outgoing traffic[^1] | Cluster, Node, Database | Total outgoing traffic (in bytes/sec) from the database |  |
+| Pending writes max | Active-Active | Maximum number of write opreations queued |  |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:47]:
```suggestion
| Pending writes max | Active-Active | Maximum number of write operations queued |  |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 5]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+|  Flash fragmentation | Database RoF, Shard RoF | Ratio between the (logical) used flash memory and the actual physical flash that is used |
+|  Flash IOPS | Cluster RoF, Node RoF | Rate of Input/Output operations per second on flash storage device |
+|  Flash KV ops | Node RoF | Rate of operations on flash key values (read + write + del) per second |
+|  Flash ops/sec | Database RoF, Shard RoF | Rate of operations on flash key values (read + write + del) per second |
+|  Free flash | Cluster RoF, Node RoF | Free space on flash storage |
+|  RAM dataset overhead | Database RoF, Shard RoF | Percentage of the RAM limit that is used for anything other than values, such as key names, dictionaries and other overheads |
+|  RAM hit ratio | Database RoF, Shard RoF | Ratio of requests processed directly from RAM to total number of requests processed |
+|  RAM limit | Database RoF | RAM limit in bytes |
+|  RAM usage | Database RoF | Percentage of the RAM limit usage |
+|  RAM:Flash access ratio | Database RoF, Shard RoF | Ratio between logical redis key value operations, and actual flash key value operations |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:81]:
```suggestion
|  RAM:Flash access ratio | Database RoF, Shard RoF | Ratio between logical Redis key value operations and actual flash key value operations |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 6]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+| Used memory | Database, Shard | Total memory used by the database, including RAM, Flash (if enabled) and replication (if enabled) | Does not include:<ol><li>Fragmentation overhead</li><li>Replica replication buffers at the primary nodes</li><li>Memory used by Lua scripts</li><li>Copy On Write (COW) operation that can be triggered by:<br><ul><li>A full replication process</li><li>A database snapshot process</li><li>AOF rewrite process</li></ul></li></ol> |
+| Write misses/sec[^1] | Database, Shard | Number of write operations (per sec) on non-existing keys |  |
 | Writes latency | Database | Latency per write operation |  |
-| Writes/sec | Database | Number of total writes per second | For example: SET |
+| Writes/sec | Database | Number of total write operation per second | For example: SET |
 
-## Redis on Flash metrics
+[^1]: These metrics are not collected during [shard migration]({{< relref "/rs/databases/configure/replica-ha" >}}). If you view the database or shard metrics while resharding, the graph will be blank.
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:60]:
```suggestion
[^1]: These metrics are not collected during [shard migration]({{< relref "/rs/databases/configure/replica-ha" >}}). If you view the database or shard metrics while resharding, the graph is blank.
```

**Review by rrelledge — COMMENTED** [review]
Here's a few small typos I noticed.

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 7]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
-| Write misses/sec | Database, Shard | Number of write operations (per sec) on non-existing keys | This metric is more relevant for caching use cases. |
+| Total keys[^1] | Database, Shard | Total number of keys in the dataset (not including replication, even if replication enabled) | Calculated as the sum of all keys of all master shards. |
+| Used memory | Database, Shard | Total memory used by the database, including RAM, Flash (if enabled) and replication (if enabled) | Does not include:<ol><li>Fragmentation overhead</li><li>Replica replication buffers at the primary nodes</li><li>Memory used by Lua scripts</li><li>Copy On Write (COW) operation that can be triggered by:<br><ul><li>A full replication process</li><li>A database snapshot process</li><li>AOF rewrite process</li></ul></li></ol> |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:55]: Should the footnote also apply to this one?

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 8]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
-| RAM fragmentation | Database, Shard | Ratio between the used (allocated) memory and the actual physical RAM that is used. |  |
-| Read misses/sec | Database, Shard | Number of read operations (per sec) on non-existing keys | This metric is more relevant for caching use cases. |
-| Reads latency | Database | Latency per read operation | The graph shows average, min, max and last values are also shown |
-| Reads/sec | Database | Number of total reads per second | For example: GET |
+| Other commands latency | Database | Latency per other command | The graph shows average, min, max and last latency values |
+| Outgoing traffic[^1] | Cluster, Node, Database | Total outgoing traffic (in bytes/sec) from the database |  |
+| Pending writes max | Active-Active | Maximum number of write opreations queued |  |
+| Pending writes min | Active-Active | Minimum number of write operations queued |  |
+| RAM fragmentation | Database, Shard | Ratio between the used (allocated) memory and the physical RAM that is used. |  |
+| Read misses/sec[^1] | Database, Shard | Number of read operations (per sec) on non-existing keys |  |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:50]:
```suggestion
| Read misses/sec[^1] | Database, Shard | Number of read operations (per sec) on nonexistent keys |  |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 9]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
-| Write misses/sec | Database, Shard | Number of write operations (per sec) on non-existing keys | This metric is more relevant for caching use cases. |
+| Total keys[^1] | Database, Shard | Total number of keys in the dataset (not including replication, even if replication enabled) | Calculated as the sum of all keys of all master shards. |
+| Used memory | Database, Shard | Total memory used by the database, including RAM, Flash (if enabled) and replication (if enabled) | Does not include:<ol><li>Fragmentation overhead</li><li>Replica replication buffers at the primary nodes</li><li>Memory used by Lua scripts</li><li>Copy On Write (COW) operation that can be triggered by:<br><ul><li>A full replication process</li><li>A database snapshot process</li><li>AOF rewrite process</li></ul></li></ol> |
+| Write misses/sec[^1] | Database, Shard | Number of write operations (per sec) on non-existing keys |  |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:56]:
```suggestion
| Write misses/sec[^1] | Database, Shard | Number of write operations (per sec) on nonexistent keys |  |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md (hunk 8 above)
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:50]:
```suggestion
| Read misses/sec[^1] | Database, Shard | Number of read operations (per sec) on nonexistent keys |  |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md (hunk 7 above)
Comment (cmilesb) [reply content/rs/clusters/monitoring/console-metrics-definitions.md]: Yes, it should. Fixed in cb911da.

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 10]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
 | Total CPU usage | Database, Shard | Percent usage of the CPU |  |
-| Total keys | Database, Shard | Total number of keys in the dataset (not including replication, even if replication enabled) | Calculated as the sum of all keys of all master shards. |
-| Used memory | Database, Shard | Total memory used by the database, including RAM, Flash (if enabled) and replication (if enabled) | **Note** – used memory does not include:<br>1. The fragmentation overhead<br><br>2. The replica replication buffers at the masters<br><br>3. Memory used by Lua scripts<br><br>4. Copy On Write (COW) operation that can be triggered by:<br><br>  - A full replication process is started<br><br>  - A database snapshot process is started<br><br>  - AOF rewrite process is started |
-| Write misses/sec | Database, Shard | Number of write operations (per sec) on non-existing keys | This metric is more relevant for caching use cases. |
+| Total keys[^1] | Database, Shard | Total number of keys in the dataset (not including replication, even if replication enabled) | Calculated as the sum of all keys of all master shards. |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:54]:
```suggestion
| Total keys[^1] | Database, Shard | Total number of keys in the dataset (not including replication, even if replication is enabled) | Calculated as the sum of all keys of all master shards. |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 11]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+| Used memory[^1] | Database, Shard | Total memory used by the database, including RAM, Flash (if enabled) and replication (if enabled) | Does not include:<ol><li>Fragmentation overhead</li><li>Replica replication buffers at the primary nodes</li><li>Memory used by Lua scripts</li><li>Copy On Write (COW) operation that can be triggered by:<br><ul><li>A full replication process</li><li>A database snapshot process</li><li>AOF rewrite process</li></ul></li></ol> |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:55]:
```suggestion
| Used memory[^1] | Database, Shard | Total memory used by the database, including RAM, Flash (if enabled), and replication (if enabled) | Does not include:<ol><li>Fragmentation overhead</li><li>Replica replication buffers at the primary nodes</li><li>Memory used by Lua scripts</li><li>Copy On Write (COW) operation that can be triggered by:<br><ul><li>A full replication process</li><li>A database snapshot process</li><li>AOF rewrite process</li></ul></li></ol> |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 12]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+| Write misses/sec[^1] | Database, Shard | Number of write operations (per sec) on non-existing keys |  |
 | Writes latency | Database | Latency per write operation |  |
-| Writes/sec | Database | Number of total writes per second | For example: SET |
+| Writes/sec | Database | Number of total write operation per second | For example: SET |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:58]:
```suggestion
| Writes/sec | Database | Number of total write operations per second | For example: SET |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 13]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
-| In Redis on Flash how come:<br>`value_in_ram` + `value_in_flash` > `total_keys` |  | Calculated as: `used_flash` / `flash_limit` |
+|  % Values in RAM | Database RoF, Shard RoF | Percent of keys whose values are stored in RAM |
+|  Flash bytes/sec | Cluster RoF, Node RoF, Database RoF, Shard RoF | Read+write bytes per second on flash storage device |
+|  Flash fragmentation | Database RoF, Shard RoF | Ratio between the (logical) used flash memory and the actual physical flash that is used |
+|  Flash IOPS | Cluster RoF, Node RoF | Rate of Input/Output operations per second on flash storage device |
+|  Flash KV ops | Node RoF | Rate of operations on flash key values (read + write + del) per second |
+|  Flash ops/sec | Database RoF, Shard RoF | Rate of operations on flash key values (read + write + del) per second |
+|  Free flash | Cluster RoF, Node RoF | Free space on flash storage |
+|  RAM dataset overhead | Database RoF, Shard RoF | Percentage of the RAM limit that is used for anything other than values, such as key names, dictionaries and other overheads |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:77]:
```suggestion
|  RAM dataset overhead | Database RoF, Shard RoF | Percentage of the RAM limit that is used for anything other than values, such as key names, dictionaries, and other overheads |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 14]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+|  RAM hit ratio | Database RoF, Shard RoF | Ratio of requests processed directly from RAM to total number of requests processed |
+|  RAM limit | Database RoF | RAM limit in bytes |
+|  RAM usage | Database RoF | Percentage of the RAM limit usage |
+|  RAM:Flash access ratio | Database RoF, Shard RoF | Ratio between logical Redis key value operations and actual flash key value operations |
+|  Used Flash | Database RoF, Shard RoF | Total RAM used to store values in Flash |
+|  Used RAM | Database RoF, Shard RoF | Total size of data stored in RAM, including keys, values, overheads and including replication, if enabled |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:83]:
```suggestion
|  Used RAM | Database RoF, Shard RoF | Total size of data stored in RAM, including keys, values, overheads, and replication (if enabled) |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 15]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+|  Flash IOPS | Cluster RoF, Node RoF | Rate of Input/Output operations per second on flash storage device |
+|  Flash KV ops | Node RoF | Rate of operations on flash key values (read + write + del) per second |
+|  Flash ops/sec | Database RoF, Shard RoF | Rate of operations on flash key values (read + write + del) per second |
+|  Free flash | Cluster RoF, Node RoF | Free space on flash storage |
+|  RAM dataset overhead | Database RoF, Shard RoF | Percentage of the RAM limit that is used for anything other than values, such as key names, dictionaries and other overheads |
+|  RAM hit ratio | Database RoF, Shard RoF | Ratio of requests processed directly from RAM to total number of requests processed |
+|  RAM limit | Database RoF | RAM limit in bytes |
+|  RAM usage | Database RoF | Percentage of the RAM limit usage |
+|  RAM:Flash access ratio | Database RoF, Shard RoF | Ratio between logical Redis key value operations and actual flash key value operations |
+|  Used Flash | Database RoF, Shard RoF | Total RAM used to store values in Flash |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:82]:
```suggestion
|  Used Flash | Database RoF, Shard RoF | Total RAM used to store values in flash |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 16]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+|  Used RAM | Database RoF, Shard RoF | Total size of data stored in RAM, including keys, values, overheads and including replication, if enabled |
+|  Values in Flash | Database RoF, Shard RoF | Number of keys with values stored in Flash, not including replication |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:84]: Off the top of my head, I'm not sure if the admin console capitalizes "flash" in the actual metric names, but the descriptions have some inconsistencies around the capitalization of flash storage.
```suggestion
|  Values in Flash | Database RoF, Shard RoF | Number of keys with values stored in flash, not including replication |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 17]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
+    avg_key_size = (ram_dataset_overhead * used_ram)
+                    / (total_keys * 2)
+    ```
+- RoF average value size in RAM
+
+    ```sh
+    avg_ram_value_size = ((1 - ram_dataset_overhead) * used_ram)    
+                        / (values_in_ram * 2)
+    ```
+- RoF average value size in Flash
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:103]:
```suggestion
- RoF average value size in flash
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 18]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
-| Incoming traffic | Cluster, Node, Database | Total incoming traffic (in bytes/sec) to the database |  |
-| Incoming traffic compressed | Active-Active | Total incoming compressed traffic (in bytes/sec) to the database |  |
-| Incoming traffic uncompressed | Active-Active | Total incoming uncompressed traffic (in bytes/sec) to the database |  |
-| Latency | Database | Latency per operation | The graph shows average, min, max and last values are also shown |
+| Hit ratio | Database, Shard | Calculated as number of operations on existing keys out of the total number of operations | `number_of_ops_on_existing_keys` / `total_ops` |
+| Incoming traffic[^1] | Cluster, Node, Database | Total incoming traffic (in bytes/sec) to the database |  |
+| Incoming traffic compressed[^1] | Active-Active | Total incoming compressed traffic (in bytes/sec) to the database |  |
+| Incoming traffic uncompressed[^1] | Active-Active | Total incoming uncompressed traffic (in bytes/sec) to the database |  |
+| Latency | Database | Latency per operation | The graph shows average, min, max and last latency values |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:39]:
```suggestion
| Latency | Database | Latency per operation | The graph shows average, min, max, and last latency values |
```

File: content/rs/clusters/monitoring/console-metrics-definitions.md [hunk 19]
```diff
@@ -18,68 +18,96 @@ aliases: /rs/administering/monitoring-metrics/definitions/
...
 | Other cmds/sec | Database | Number of other commands per second | For example: PING, AUTH, INFO |
-| Other commands latency | Database | Latency per other command | The graph shows average, min, max and last values are also shown |
-| Outgoing traffic | Cluster, Node, Database | Total outgoing traffic (in bytes/sec) from the database |  |
-| Pending writes max | Active-Active | Maximum number of writes queued |  |
-| Pending writes min | Active-Active | Minimum number of writes queued |  |
-| RAM fragmentation | Database, Shard | Ratio between the used (allocated) memory and the actual physical RAM that is used. |  |
-| Read misses/sec | Database, Shard | Number of read operations (per sec) on non-existing keys | This metric is more relevant for caching use cases. |
-| Reads latency | Database | Latency per read operation | The graph shows average, min, max and last values are also shown |
-| Reads/sec | Database | Number of total reads per second | For example: GET |
+| Other commands latency | Database | Latency per other command | The graph shows average, min, max and last latency values |
```
Comment (rrelledge) [inline content/rs/clusters/monitoring/console-metrics-definitions.md:45]:
```suggestion
| Other commands latency | Database | Latency per other command | The graph shows average, min, max, and last latency values |
```

**Review by rrelledge — APPROVED** [review]
Looks good to me. I just left a few more suggestions.

**Pull request merged**